.github/workflows/ci.yml

```diff
@@ -24,36 +24,36 @@ jobs:
         CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ./ci/do_ci.sh cmake.test
 
   cmake_gcc_maintainer_test:
-    name: CMake gcc (maintainer mode)
+    name: CMake gcc 12 (maintainer mode)
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
       with:
         submodules: 'recursive'
     - name: setup
       run: |
-        sudo CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ./ci/setup_cmake.sh
-        sudo CC=/usr/bin/gcc-10 CXX=/usr/bin/g++-10 ./ci/setup_ci_environment.sh
+        sudo CC=/usr/bin/gcc-12 CXX=/usr/bin/g++-12 GOOGLETEST_VERSION=1.12.1 ./ci/setup_cmake.sh
```

Comment:
If gcc-12 is the default compiler with ubuntu-latest, do we need to set it explicitly here ?

Reply:
The installed GCC compilers are: GNU C++ 9.5.0, 10.4.0, 11.3.0, 12.1.0.

Which one is the default, I don't know.

Naming explicitly the compiler we want is more robust and avoids surprises, should this list change again with a different ubuntu-latest image.